### ISSUE TITLE
ES6 class setters and getters do not work for field in schema

### ISSUE THREAD

**Issue by mikefowler** [sequelize/sequelize#9300]
## What are you doing?
Attempting to use a custom setter on my model class. 

The Sequelize release notes for 4.x, under "[New Features](http://docs.sequelizejs.com/manual/tutorial/upgrade-to-v4.html#new-features)", reads: "You can then define custom methods, class methods and **getters/setter** directly in the class".

```js
// Foo.js

import Sequelize, { Model } from 'sequelize';

class Foo extends Model {
  get languages() {
    console.log('Getting languages…');
    return JSON.parse(this.getDataValue('languages'));
  }

  set languages(value) {
    console.log('Setting languages to', value);
    this.setDataValue(JSON.stringify(value));
  }
}

const SCHEMA = {
  languages: Sequelize.TEXT,
};

export default (sequelize) => {
  Foo.init(SCHEMA, { sequelize });
}

// index.js

foo.update({
  languages: ['en', 'es'],
});
```

## What do you expect to happen?
- `Setting languages to ['en', 'es']` is logged to the console
- The value `"["en","es"]"` is persisted to the database

## What is actually happening?
- The setter method is not being called, and as a result, Sequelize throws a validation error: `SequelizeValidationError: string violation: languages cannot be an array or an object`

__Dialect:__ mysql (but irrelevant, because the error is thrown before reaching the adapter)
__Dialect version:__ n/a
__Database version:__ n/a
__Sequelize version:__ 4.22.0
__Tested with latest release:__ No

**Note :** _Your issue may be ignored OR closed by maintainers if it's not tested against latest version OR does not follow issue template._

**Comment by heisian** [comment]
I think there's an error in your code:
```node
  set languages(value) {
    console.log('Setting languages to', value);

    // this.setDataValue(JSON.stringify(value)); <-- MISSING KEY TO SET

    this.setDataValue('languages', JSON.stringify(value));
  }
```

http://docs.sequelizejs.com/class/lib/model.js~Model.html#instance-method-setDataValue

**Comment by mikefowler** [comment]
@heisian yes, that seems to be a typo (which I've now corrected in my psuedocode), but the issue is that the setters and getters are not being called, period. In a simple test test, those `console.log` statements are never run.

**Comment by jamesbrucepower** [comment]
I currently have exactly the same issue, after spending much time debugging models.js source I gave up and added VIRTUAL attributes with getters and setters

**Comment by sushantdhiman** [comment]
Looks like we are still using `options.getterMethods / options.setterMethods`, I thought we remove support for that with v4.

Tagged as bug

**Comment by 0x62** [comment]
Any update on this?

**Comment by toymachiner62** [comment]
My setter does not seem to be called either using sequelize 6
